src/effects/native/phasereffect.cpp

```diff
@@ -0,0 +1,199 @@
+    range->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    range->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    range->setDefault(0.5);
+    range->setMinimum(0.05);
+    range->setMaximum(0.95);
+
+    return manifest;
+}
+
+PhaserEffect::PhaserEffect(EngineEffect* pEffect,
+                           const EffectManifest& manifest) 
+        : m_pStagesParameter(pEffect->getParameterById("stages")),
+          m_pLFOFrequencyParameter(pEffect->getParameterById("lfo_frequency")),
+          m_pDepthParameter(pEffect->getParameterById("depth")),
+          m_pFeedbackParameter(pEffect->getParameterById("feedback")),
+          m_pRangeParameter(pEffect->getParameterById("range")),
+          m_pStereoParameter(pEffect->getParameterById("stereo")) {
+    Q_UNUSED(manifest);
+}
+
+PhaserEffect::~PhaserEffect() {
+    //qDebug() << debugString() << "destroyed";
+}
+
+void PhaserEffect::processChannel(const ChannelHandle& handle,
+                                  PhaserGroupState* pState,
+                                  const CSAMPLE* pInput, CSAMPLE* pOutput,
+                                  const unsigned int numSamples,
+                                  const unsigned int sampleRate,
+                                  const EffectProcessor::EnableState enableState,
+                                  const GroupFeatureState& groupFeatures) {
+
+    Q_UNUSED(handle);
+    Q_UNUSED(enableState);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(sampleRate);
+
+    CSAMPLE frequency = m_pLFOFrequencyParameter->value();
+    CSAMPLE depth = m_pDepthParameter->value();
+    CSAMPLE feedback = m_pFeedbackParameter->value();
+    CSAMPLE range = m_pRangeParameter->value();
+    int stages = m_pStagesParameter->value();
+
+    CSAMPLE* oldInLeft = pState->oldInLeft;
+    CSAMPLE* oldOutLeft = pState->oldOutLeft;
+    CSAMPLE* oldInRight = pState->oldInRight;
+    CSAMPLE* oldOutRight = pState->oldOutRight;
+
+    //Using two sets of coefficients for left and right channel 
+    CSAMPLE filterCoefLeft = 0;
+    CSAMPLE filterCoefRight = 0;
+
+    //If changing the 'Rate' knob, the phaser will use the initial parameters    
+    //until a desired frequency is chosen (to avoid bubbling noises)
+    if (frequency != pState->frequency) {
+        pState->wait = 0;
+        pState->frequency = frequency; 
+        pState->change = true;
+    }
+
+    if (pState->change) {
+        pState->wait++;
+        if (pState->wait == pState->limit) {
+            pState->change = false;
+            pState->oldFrequency = frequency;
+        } else {
+            frequency = pState->oldFrequency;
+        }
+    }
+
+    CSAMPLE left = 0, right = 0;
+    CSAMPLE leftPhase, rightPhase;
+    CSAMPLE freqSkip = frequency * 2.0 * M_PI / sampleRate;
+
+    int stereoCheck = m_pStereoParameter->value();
+    int counter = 0;
+
+    const int kChannels = 2;
+    for (unsigned int i = 0; i < numSamples; i += kChannels) {
+        pState->time++;
```

Comment:
Time is growing infinite and will finally overflow. You should store the phase instead and add a phase fraction per sample. This will also avoid the bubbling noise.